### DISCUSSION TITLE
in compose() waiting for yield

### DISCUSSION THREAD

**Discussion by leemthai** [Textualize/textual#4853] [Q&A] [answered]
This probably makes no sense because I don't understand async programming at all but here goes anyway. Here is the code:

```
def compose(self) -> ComposeResult:
        yield CoinTable(self.x)
        yield CoinInput()
```

Is there a way to wait for CoinTable() and CoinInput() widgets to yield before running any more code in Compose?

I want to yield further widgets but they really need CoinTable() to be available (in the DOM?) as they read values from CoinTable() when initialzing...

Thanks

**Comment by willmcgugan** [comment, marked as answer]
It is best to implement reading data in your `Mount` event handler, rather than `__init__`. That way the DOM is guaranteed to be available.

i.e. 

```
def on_mount(self):
    ...
```
